samples/faceDetection.js

```diff
@@ -33,93 +33,82 @@ const fs = require('fs');
-      console.error('ERROR:', err);
-      callback(err);
-    });
+
+  const [{faceAnnotations: faces}] = await client.faceDetection(request);
+  const {length: numFaces} = faces;
+
+  console.log('Found ' + numFaces + (numFaces === 1 ? ' face' : ' faces'));
+  return faces;
 }
 // [END vision_face_detection_tutorial_send_request]
 
 /**
  * Draws a polygon around the faces, then saves to outputFile.
  */
 // [START vision_face_detection_tutorial_process_response]
-function highlightFaces(inputFile, faces, outputFile, Canvas, callback) {
-  fs.readFile(inputFile, (err, image) => {
-    if (err) {
-      return callback(err);
-    }
-
-    const Image = Canvas.Image;
-    // Open the original image into a canvas
-    const img = new Image();
-    img.src = image;
-    const canvas = new Canvas(img.width, img.height);
-    const context = canvas.getContext('2d');
-    context.drawImage(img, 0, 0, img.width, img.height);
-
-    // Now draw boxes around all the faces
-    context.strokeStyle = 'rgba(0,255,0,0.8)';
-    context.lineWidth = '5';
-
-    faces.forEach(face => {
-      context.beginPath();
-      let origX = 0;
-      let origY = 0;
-      face.boundingPoly.vertices.forEach((bounds, i) => {
-        if (i === 0) {
-          origX = bounds.x;
-          origY = bounds.y;
-        }
-        context.lineTo(bounds.x, bounds.y);
-      });
-      context.lineTo(origX, origY);
-      context.stroke();
-    });
+async function highlightFaces(inputFile, faces, outputFile, Canvas) {
+  const image = fs.readFileSync(inputFile);
 
-    // Write the result to a file
-    console.log('Writing to file ' + outputFile);
-    const writeStream = fs.createWriteStream(outputFile);
-    const pngStream = canvas.pngStream();
+  const Image = Canvas.Image;
 
-    pngStream.on('data', chunk => {
-      writeStream.write(chunk);
+  // Open the original image into a canvas
+  const img = new Image();
+  img.src = image;
+  const canvas = new Canvas(img.width, img.height);
+  const context = canvas.getContext('2d');
+  context.drawImage(img, 0, 0, img.width, img.height);
+
+  // Now draw boxes around all the faces
+  context.strokeStyle = 'rgba(0,255,0,0.8)';
+  context.lineWidth = '5';
+
+  faces.forEach(face => {
+    context.beginPath();
+    let origX = 0;
+    let origY = 0;
+    face.boundingPoly.vertices.forEach((bounds, i) => {
+      if (i === 0) {
+        origX = bounds.x;
+        origY = bounds.y;
+      }
+      context.lineTo(bounds.x, bounds.y);
     });
-    pngStream.on('error', console.log);
-    pngStream.on('end', callback);
+    context.lineTo(origX, origY);
+    context.stroke();
+  });
+
+  // Write the result to a file
+  console.log('Writing to file ' + outputFile);
+  const writeStream = fs.createWriteStream(outputFile);
+  const pngStream = canvas.pngStream();
+
+  pngStream.on('data', chunk => {
+    writeStream.write(chunk);
+  });
+  pngStream.on('error', console.log);
+  pngStream.on('end', () => {
+    return;
```

Comment:
Sadly that ain't going to work :) If we want this to be an async function, you'll need to wrap this whole thing:

```js
async function highlightFaces(inputFile, faces, outputFile, Canvas) {
  return new Promise((resolve, reject) => {
    const image = ....
    // other code n'stuff
    pngStream
      .on('error', err => { 
        console.error(err);
        reject(err);
       })
      .on('end', () => resolve());
```

Or something like that